packages/browser-client/src/Components/PeerButtons.tsx

```diff
@@ -24,360 +24,418 @@ import {
   reassembleBlock,
   shortId,
 } from 'portalnetwork'
-import React, { Dispatch, SetStateAction, useContext, useState } from 'react'
-import { BlockContext, HistoryProtocolContext, PeersContext } from '../ContextHooks'
+import React, { useContext, useEffect, useReducer } from 'react'
+import { AppContext, StateChange } from '../globalReducer'
 
-export interface PeerButtonsProps {
-  peerIdx: number
-  setPeerIdx: Dispatch<SetStateAction<number>>
-  sortedDistList: [number, string[]][]
-  peer: ENR
-  // peers: ENR[]
+interface PeerButtonsState {
+  epoch: number
+  offer: Uint8Array[]
+  ping: [string, string]
+  distance: string
+  blockHash: string
 }
 
-export default function PeerButtons(props: PeerButtonsProps) {
-  const { setBlock } = useContext(BlockContext)
-  const peers = useContext(PeersContext)
-  const [epoch, setEpoch] = useState(0)
-  const { peerIdx, sortedDistList, peer } = props
-  const [offer, setOffer] = useState<Uint8Array[]>([])
-  const [pinging, setPinging] = useState(false)
-  const [ponged, setPonged] = useState<boolean | undefined>()
-  const [distance, setDistance] = useState('')
-  const [blockHash, setBlockHash] = useState('')
-  const historyProtocol = useContext(HistoryProtocolContext)
-  const addToOffer = (type: string) => {
-    const contentKeys: Uint8Array[] = []
+const peerButtonsInitialState = {
+  epoch: 0,
+  offer: [],
+  ping: ['blue.100', 'PING'] as [string, string],
+  distance: '',
+  blockHash: '',
+}
+
+enum PeerButtonsStateChange {
+  SETEPOCH = 'SETEPOCH',
+  ADDTOOFFER = 'ADDTOOFFER',
+  PING = 'PING',
+  SETDISTANCE = 'SETDISTANCE',
+  SETBLOCKHASH = 'SETBLOCKHASH',
+  SETPEERIDX = 'SETPEERIDX',
+}
+
+interface AppStateAction {
+  type: PeerButtonsStateChange
+  payload?: any
+}
+
+const reducer = (state: PeerButtonsState, action: AppStateAction) => {
```

Comment:
Can we put this reducer in a separate utility file and keep all this state management logic out of the front-end?  I don't want to overengineer this but since we're baking all this stuff in, it would be nice to have the reducers and state management logic all in their own subdirectory.